From b95374d913fabe27be4b92262a8f8bc9d3897f88 Mon Sep 17 00:00:00 2001
From: ryanking13 <def6488@gmail.com>
Date: Sat, 20 Jan 2024 19:02:32 +0900
Subject: [PATCH 4/6] Raise when no argument is given

Emscripten 3.1.51 does not raise an error when no argument is given.
Some build tools (e.g. ffmpeg) relies on this behavior, so we should
keep it.

Upstream issue: https://github.com/emscripten-core/emscripten/issues/21116
---
 emcc.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/emcc.py b/emcc.py
index e546629c3..52af8b9e1 100644
--- a/emcc.py
+++ b/emcc.py
@@ -640,6 +640,9 @@ There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR P
       print(libname)
     return 0
 
+  if not input_files and not state.link_flags:
+    exit_with_error('no input files')
+
   if options.reproduce:
     create_reproduce_file(options.reproduce, args)
 
-- 
2.34.1

